src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1589,7 +1626,7 @@ private static bool WrongArity(Symbol symbol, int arity, bool diagnose, LookupOp
                         NamedTypeSymbol namedType = (NamedTypeSymbol)symbol;
                         // non-declared types only appear as using aliases (aliases are arity 0)
                         Debug.Assert(object.ReferenceEquals(namedType.ConstructedFrom, namedType));
-                        if (namedType.Arity != arity || options.IsAttributeTypeLookup() && arity != 0)
+                        if (namedType.Arity != arity)
```

Comment:
I'm worried about another scenario:
Consider various nested scopes (container types), we could have a generic type in a more nested scope (that type would have previously been ignored) and a valid type in an outer scope.
The change to `WrongArity` means that the generic type in inner scope is going to be considered viable, which means that the non-generic type in outer scope will be ignored :-(

I'm thinking we either need to 
1. return to your original approach (`CheckViability` and `WrongArity` consider LangVer)
2. we need to do two lookups: one ignoring generic types (as before) and one considering them (if the first lookup didn't yield a result). Then we can do LangVer check and produce a diagnostic if needed. This design guarantees that results that were found before are still found first.

@AlekseyTs Could I ask you opinion on this?
I initially pushed back on having lookup logic depend on LangVer, but the alternative seems to be doing two lookups. Which one is better? Is there some other alternative?